boot/freeldr/freeldr/arch/realmode/fathelp.inc

```diff
@@ -61,12 +65,28 @@ start:
 
 
 FatHelperEntryPoint:
+    /* Clear the segment values of the indirect far calls to make sure it works */
+    xor bx, bx
+    mov word ptr ss:[bp-(ReadSectorsOffset-2)], bx
+    mov word ptr ss:[bp-(ReadClusterOffset-2)], bx
+    mov word ptr ss:[bp-(PutCharsOffset-2)], bx
```

Comment:
This could be moved here:
https://github.com/reactos/reactos/pull/7530/files#diff-27faec1c6232247683095c94ff55f6d45e5114ebac69b0a6cdc7ea8fc369ab65R223
and there:
https://github.com/reactos/reactos/pull/7530/files#diff-36a3a50dd173dcde3f66fcbc459bd3778412dda5f354425abed40ec0554a7a9aR249
so that when the fat.S (or pc98/fat12fdd.S) code jumps to freeldr, the functions offsets and segments are correctly set already.

Reply:
but i don't have enough space in the bootsector

Reply:
So what I wonder is, whether you would have enough space in the bootsector itself if you were to do:
```asm
    mov dword ptr [bp-ReadSectorsOffset], offset ReadSectorsF    // Save the address of ReadSectorsF
    mov dword ptr [bp-ReadClusterOffset], offset ReadClusterF    // Save the address of ReadClusterF
    mov dword ptr [bp-PutCharsOffset],    offset PutCharsF       // Save the address of PutCharsF
```
(instead of using the `word ptr` as currently done) and whether it would still work.
This would, obviously, assume that the offset proper is still < 0xFFFF .